CMakeLists.txt

```diff
@@ -91,16 +94,20 @@ if(CMAKE_COMPILER_IS_CLANG)
     set(CMAKE_C_FLAGS_CHECK       "-Werror -Os")
 endif(CMAKE_COMPILER_IS_CLANG)
 
-if(MSVC)
+if(CMAKE_COMPILER_IS_IAR)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --warn_about_c_style_casts --warnings_are_errors -Ohz")
```

Comment:
Have you considered adding "--strict"? Or does the compilation fail with this option?
Did you check if you have to use "--warnings_affect_exit_code" too, or is the "--warnings_are_errors" sufficient?

Reply:
I tried these options and found that ```--strict``` conflicts with default ```-e              Enable IAR C/C++ language extensions```. ```--warnings_are_errors``` covers ```--warnings_affect_exit_code```.